src/managers/GuildBanManager.js

```diff
@@ -0,0 +1,167 @@
+const GuildBan = require('../structures/GuildBan');
+const GuildMember = require('../structures/GuildMember');
+const Collection = require('../util/Collection');
+
+/**
+ * Manages API methods for Guilds bans and stores their cache.
+ * @extends {BaseManager}
+ */
+class GuildBanManager extends BaseManager {
+  constructor(guild, iterable) {
+    super(guild.client, iterable, GuildBan);
+
+    /**
+     * The guild this Manager belongs to
+     * @type {Guild}
+     */
+    this.guild = guild;
+  }
+
+  /**
+   * The cache of this Manager
+   * @type {Collection<Snowflake, GuildBan>}
+   * @name GuildBanManager#cache
+   */
+
+  add(data, cache) {
+    return super.add(data, cache, { id: data.user.id, extras: [this.guild] });
+  }
+
+  /**
+   * Data that resolves to give a GuildBan object. This can be:
+   * * A GuildBan object
+   * * A User resolvable
+   * @typedef {GuildBan|UserResolvable} GuildBanResolvable
+   */
+
+  /**
+   * Resolves a GuildBanResolvable to a GuildBan object.
+   * @param {GuildBanResolvable} ban The ban that is in the guild
+   * @returns {?GuildBan}
+   */
+  resolve(ban) {
+    const banResolvable = super.resolve(ban);
+    if (banResolvable) return banResolvable;
+    const userResolvable = this.client.users.resolveID(ban);
+    if (userResolvable) return super.resolve(userResolvable);
+    return null;
+  }
+
+  /**
+   * Options used to fetch a single ban from a guild.
+   * @typedef {Object} FetchBanOptions
+   * @property {UserResolvable} user The ban to fetch
+   * @property {boolean} [cache=true] Whether or not to cache the fetched ban
+   * @property {boolean} [force=false] Whether to skip the cache check and request the API
+   */
+
+  /**
+   * Fetches ban(s) from Discord.
+   * @param {UserResolvable|FetchBanOptions} [options] If provided fetches a single ban.
+   * If undefined, fetches all bans.
+   * @returns {Promise<GuildBan>|Promise<Collection<Snowflake, GuildBan>>}
+   * @example
+   * // Fetch all bans from a guild
+   * guild.bans.fetch()
+   *   .then(console.log)
+   *   .catch(console.error);
+   * @example
+   * // Fetch a single ban
+   * guild.bans.fetch('351871113346809860')
+   *   .then(console.log)
+   *   .catch(console.error);
+   * @example
+   * // Fetch a single ban without checking cache
+   * guild.bans.fetch({ user, force: true })
+   *   .then(console.log)
+   *   .catch(console.error)
+   * @example
+   * // Fetch a single ban without caching
+   * guild.bans.fetch({ user, cache: false })
+   *   .then(console.log)
+   *   .catch(console.error);
+   */
+  fetch(options) {
+    if (!options) return this._fetchMany();
+    const user = this.client.users.resolveID(options);
+    if (user) return this._fetchSingle({ user, cache: true });
+    if (options.user) {
+      options.user = this.client.users.resolveID(options.user);
+    }
+    if (!options.user) {
+      if ('cache' in options) return this._fetchMany(options.cache);
+      throw new Error('FETCH_BAN_RESOLVE_ID');
```

Comment:
```suggestion
      Promise.reject(new Error('FETCH_BAN_RESOLVE_ID'));
```
the method isn't async, so you can't use throw here